dev/tools/localization/gen_l10n_types.dart

```diff
@@ -374,19 +374,37 @@ class AppResourceBundle {
     }
 
     String localeString = resources['@@locale'] as String;
+
+    // If the locale string was not defined in @@locale, look for the first
+    // instance of an ISO 639-1 language code, matching exactly.
     if (localeString == null) {
-      final RegExp filenameRE = RegExp(r'^[^_]*_(\w+)\.arb$');
-      final RegExpMatch match = filenameRE.firstMatch(file.path);
-      localeString = match == null ? null : match[1];
+      final String fileName = file.path
```

Comment:
This will not work on Windows. Best to use the path package https://pub.dev/documentation/path/latest/path/basenameWithoutExtension.html